File: sizeofpalindrome.cpp
#include<iostream>
#include<map>
#include<string>
using namespace std;

int main(){
    string s="aaaabbbbcccdde";
    int count=0,flag=0;
    map<char,int>m;
    for(int i=0;i<s.size();i++)
    m[s[i]]++;
    for(auto i:m){
        if(i.second%2==0)
        count+=i.second;
        else{
            count+=i.second-1;
            flag=1;
        }
    }
    cout<<count+flag;
    return 0;
}

File: sizeofpalindrome.java
import java.util.HashMap;

public class sizeofpalindrome{
    public static void main(String[] args) {
        String s = "aaaabbbbcccdde";
        int count = 0, flag = 0;
        HashMap<Character, Integer> m = new HashMap<>();

            // Count character frequencies
            for (int i = 0; i < s.length(); i++) {
                char ch = s.charAt(i);
                m.put(ch, m.getOrDefault(ch, 0) + 1);
            }
    
            // Calculate palindrome length
            for (int freq : m.values()) {
                if (freq % 2 == 0)
                    count += freq;
                else {
                    count += freq - 1;
                    flag = 1;
                }
            }
    
            System.out.println(count + flag);
        }
    }